src/ophyd_async/epics/areadetector/controllers/ad_controller.py

```diff
@@ -0,0 +1,42 @@
+import asyncio
+from typing import Optional
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    AsyncStatus,
+    DetectorControl,
+    DetectorTrigger,
+    set_and_wait_for_value,
+    wait_for_value,
+)
+
+from ..drivers.ad_driver import ADDriver, ImageMode
+
+
+class ADController(DetectorControl):
+    def __init__(self, drv: ADDriver) -> None:
+        self.driver = drv
+
+    def get_deadtime(self, exposure: float) -> float:
+        return 0.002
```

Comment:
should this return Optional[float]?

Reply:
resolution; This is only used for fly scanning, at which point a fly scannable detector must have some deadtime, and it cannot be 0.

Reply:
We could raise an exception to make it clear that a `SimDetector` cannot be hardware triggered